sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java

```diff
@@ -102,4 +106,29 @@ private static Long parseToEpochSeconds(String expiresOn, String expiresIn) {
         }
         throw LOGGER.logExceptionAsError(new IllegalArgumentException("Unable to parse date time " + dateToParse));
     }
+
+    private static OffsetDateTime parseRefreshIn(String refreshIn) {
+        if (!CoreUtils.isNullOrEmpty(refreshIn)) {
+            try {
+                long seconds = Long.parseLong(refreshIn);
+                return OffsetDateTime.now(ZoneOffset.UTC).plusSeconds(seconds);
+            } catch (NumberFormatException e) {
+                throw LOGGER.logExceptionAsError(
+                    new IllegalArgumentException("Unable to parse refresh in  " + refreshIn, e));
+            }
+
+        }
+        return null;
+    }
+
+    private static OffsetDateTime inferManagedIdentityRefreshInValue(OffsetDateTime expiresOn) {
+        OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
+
+        if (expiresOn.isAfter(now.plus(Duration.ofHours(2))) && expiresOn.isBefore(OffsetDateTime.MAX)) {
```

Comment:
how would `expiresOn` not be before `MAX`?

Reply:
Just a sanity check ported from .NET impl.

Reply:
Probably can be removed since an OffsetDateTime defaults to null, so it seems pretty unlikely it would somehow be MAX..